R/eml.R

```diff
@@ -29,6 +29,61 @@ pid_to_eml_other_entity <- function(mn, pids) {
   sysmeta_to_eml_other_entity(sysmeta)
 }
 
+#' Create an EML dataTable object for a given PID. 
+#' This function generates an attributeList and physical and constructs a dataTable.
+#'
+#' @param mn (MNode) Member Node where the PID is associated with an object.
+#' @param pid (character) The PID of the object to create the dataTable for.
+#' @param attributes (data.frame) Data frame of attributes.
+#' @param factors (data.frame) Data frame of enumerated attribute values (factors).
+#' @param name (character) Optional field to specify entityName, otherwise will be extracted from system metadata.
+#' @param description (character) Optional field to specify entityDescription, otherwise will match name.
+#' 
+#' @return (dataTable) The dataTable object
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' # Generate a dataTable for a given pid
+#' attributes <- create_dummy_attributes_dataframe(10)
+#' name <- "1234.csv"
+#' description <- "A description of this entity."
+#' dataTable <- pid_to_eml_datatable(mn, pid, attributes, name=name, description=description)
+#' }
+pid_to_eml_datatable <- function(mn, pid, attributes, factors=NULL, name=NULL, description=NULL) {
```

Comment:
Add spaces around named args (`factors = NULL`). If this makes the line longer than 80 columns, format it like:

```r
pid_to_eml_datatable <- function(mn, 
                                 pid, 
                                 attributes, 
                                 factors = NULL, 
                                 name = NULL, 
                                 description = NULL) {
```